### PR TITLE
Zoltan2 Sphynx: Add options for initial guess and full/partial ortho

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/zoltan2 

## Motivation
Add option for a blockwise constant initial guess and full/partial orthogonalization.

### CODE REVIEW

File: packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp [hunk 1]
```diff
@@ -558,14 +563,41 @@ namespace Zoltan2 {
     anasaziParams.set("Orthogonalization", ortho);
     anasaziParams.set("Use Locking", lock);
     anasaziParams.set("Relative Locking Tolerance", relLockTol);
+    anasaziParams.set("Full Ortho", useFullOrtho);
 
+    typedef typename Adapter::user_t graph_type;
+    typedef typename graph_type::global_ordinal_type GO;
```
Comment (seheracer) [inline packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp:569]: Global ordinal is already defined as gno_t in line 105.  Do we need GO?

**Comment by seheracer** [comment]
Thanks, @cgcgcg! Changes look good for now 👍

File: packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp (hunk 1 above)
Comment (seheracer) [reply packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp]: Resolved, thanks!

Comment (cgcgcg) [reply packages/zoltan2/sphynx/src/Zoltan2_Sphynx.hpp]: No, we don't :+1:

**Pull request merged**